### PR TITLE
Add startup method wrapper

### PR DESCRIPTION
**Pull request by puiterwijk**
Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -115,6 +115,16 @@ impl Context {
         }
     }
 
+    pub fn startup(&mut self, startup_type: TPM2_SU) -> Result<()> {
+        let ret = unsafe { Esys_Startup(self.mut_context(), startup_type) };
```
Comment (Superhepper) [inline src/context.rs:119]: There is a Startup type already defined in src/constants/types.rs so you do not need to use the ESYS type in the api.

Comment (puiterwijk) [reply src/context.rs:119]: Ah, right. But this function is going to be troublesome with the tests anyway (given that swtpm already runs a startup, and doesn't like having it rerun, as a normal TPM should), so I may very well just drop this PR.

Comment (ionut-arm) [reply src/context.rs:119]: We do actually have to run the CLI version of this command before running tests: https://github.com/parallaxsecond/rust-tss-esapi/blob/master/tests/all.sh#L18 . Plus, even without a test, the method might be useful on `Context`

Comment (Superhepper) [reply src/context.rs:119]: Yeah, I do not think it should be closed. I think this method needs to be available even without a test.

**Comment by puiterwijk** [comment]
By popular demand, reopened this PR (sorry about closing it).
I rebased and made it use the `StartupType` as requested.

**Review by ionut-arm — APPROVED** [review]
Nice and simple

File: src/context.rs [hunk 2]
```diff
@@ -130,6 +130,16 @@ impl Context {
...
     }
 
+    pub fn startup(&mut self, startup_type: StartupType) -> Result<()> {
+        let ret = unsafe { Esys_Startup(self.mut_context(), startup_type.into()) };
+        let ret = Error::from_tss_rc(ret);
+        if ret.is_success() {
+            Ok(())
+        } else {
+            Err(ret)
+        }
```
[the remark this answers is not in the document]
Comment (puiterwijk) [reply src/context.rs]: Added!

**Review by Superhepper — APPROVED** [review]

**Pull request merged**